src/timely-util/src/operator.rs

```diff
@@ -143,6 +144,10 @@ where
     /// Take a Timely stream and convert it to a Differential stream, where each diff is "1"
     /// and each time is the current Timely timestamp.
     fn pass_through<R: Data>(&self, name: &str, unit: R) -> Stream<G, (D1, G::Timestamp, R)>;
+
+    /// Wraps the stream with a passthrough operator that will be shut down when the provided
+    /// token cannot be upgraded anymore.
+    fn fused(&self, token: Weak<()>) -> Stream<G, D1>;
```

Comment:
I'm not in favor of the name `fused`. I think the function should represent a verb, not a noun, and fused mostly means ["some things mixed together,"](https://www.merriam-webster.com/dictionary/fuse) which is not the meaning we want to have here. I liked the old `with_token` (?) better!

Reply:
The documentation can be more specific to what the implementation actually provides. At the moment, it will gate all data once the `token` is dropped, but still passes through progress traffic. For some dataflows, this might be surprising if they assume that dropping the token also advances the frontier to the empty frontier.

Reply:
I like having "fuse" in the name because it evokes the notion of a thing that stops stuff from flowing through once it is triggered. (It would be nicer to have a water analogy rather than one from electricity, but I can't think of one.) "token" doesn't evoke the same associations for me, it's pretty generic. But it does have a specific meaning in the context of MZ dataflows, so I think it would be fine too.

A couple more entries to the bike shedding:
* `with_fuse`
* `with_circuit_breaker`

Reply:
The noun form of fuse is distinct from the verb form here, so I prefer `with_fuse` over `fused`.  Circuit breakers normally can be reset, unlike fuses, so I'd prefer to avoid that here.